site/src/main/java/uk/nhs/digital/ps/beans/Publication.java

```diff
@@ -217,23 +217,23 @@ public boolean isPubliclyAccessible() {
     }
 
     /*
-     * Migration: 20171208
-     * New field: publicationsystem:Attachments-v2
-     * Old field: publicationsystem:attachments
+     * Migration: 20171213
+     * New field: publicationsystem:Attachments-v3
+     * Old field: publicationsystem:Attachments-v2
      *
-     * Temporary make this getter forward and backward compatible, meaning we can deploy this code without running
-     * content migration.
+     * Temporary supporting old and new versions until all the live content is migrated to the new way
      */
+    @HippoEssentialsGenerated(internalName = PropertyKeys.ATTACHMENTS_V3)
+    public List<Attachment> getAttachments() {
+        assertPropertyPermitted(PropertyKeys.ATTACHMENTS_V3);
+
+        return getChildBeansIfPermitted(PropertyKeys.ATTACHMENTS_V3, Attachment.class);
+    }
     @HippoEssentialsGenerated(internalName = PropertyKeys.ATTACHMENTS_V2)
-    public List<HippoResourceBean> getAttachments() {
+    public List<HippoResourceBean> getAttachmentsOld() {
         assertPropertyPermitted(PropertyKeys.ATTACHMENTS_V2);
-        List<HippoResourceBean> attachments = getChildBeansByName(PropertyKeys.ATTACHMENTS_V2, HippoResourceBean.class);
-
-        if (attachments.isEmpty()) {
-            attachments = getChildBeansByName(PropertyKeys.ATTACHMENTS, HippoResourceBean.class);
-        }
 
-        return attachments;
+        return getChildBeansByName(PropertyKeys.ATTACHMENTS_V2, HippoResourceBean.class);
```

Comment:
Using `getChildBeansIfPermitted` would eliminate the need to call `assertPropertyPermitted` above.

Reply:
👍